acceptance-tests/src/test-support/java/tech/pegasys/artemis/test/acceptance/dsl/ArtemisNode.java

```diff
@@ -0,0 +1,187 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.provider.JsonProvider;
+import tech.pegasys.artemis.test.acceptance.dsl.data.BeaconHead;
+
+public class ArtemisNode {
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String ARTEMIS_BINARY =
+      new File(System.getProperty("artemis.binary.path", "../build/install/artemis/bin/artemis"))
+          .getAbsolutePath();
+
+  private final SimpleHttpClient httpClient;
+
+  private boolean started = false;
+  private File configFile;
+  private Process process;
+  private Path workDir;
+
+  ArtemisNode(final SimpleHttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  public void start() throws Exception {
+    assertThat(started).isFalse();
+    started = true;
+    workDir = Files.createTempDirectory("artemis");
+    configFile = File.createTempFile("config", ".toml");
+    configFile.deleteOnExit();
+    new ConfigBuilder().writeTo(configFile);
+    LOG.info(
+        "Starting artemis from {} with config file {}",
+        ARTEMIS_BINARY,
+        configFile.getAbsolutePath());
+    process =
+        new ProcessBuilder(
+                ARTEMIS_BINARY, "--config", configFile.getAbsolutePath(), "--logging", "DEBUG")
+            .directory(workDir.toFile())
+            .inheritIO()
+            .start();
+  }
+
+  public void waitForGenesis() {
+    waitFor(() -> httpClient.get(getRestApiUrl(), "/node/genesis_time"));
+  }
+
+  public void waitForNewBlock() throws IOException {
+    final Bytes32 startingBlockRoot = getCurrentBeaconHead().getBlockRoot();
+    waitFor(
+        () -> assertThat(getCurrentBeaconHead().getBlockRoot()).isNotEqualTo(startingBlockRoot));
+  }
+
+  private BeaconHead getCurrentBeaconHead() throws IOException {
+    final BeaconHead beaconHead =
+        JsonProvider.jsonToObject(
+            httpClient.get(getRestApiUrl(), "/beacon/head"), BeaconHead.class);
+    LOG.debug("Retrieved beacon head: {}", beaconHead);
+    return beaconHead;
+  }
+
+  private URI getRestApiUrl() {
+    return URI.create("http://127.0.0.1:9051");
+  }
+
+  public void stop() {
+    if (!started) {
+      return;
+    }
+    LOG.debug("Shutting down");
+    try {
+      process.destroy();
+      if (!process.waitFor(1, TimeUnit.MINUTES)) {
+        LOG.warn("Didn't shutdown in a timely fashion, being more aggressive");
+        process.destroyForcibly();
+      }
+
+      if (!configFile.delete() && configFile.exists()) {
+        throw new IOException("Failed to delete config file: " + configFile);
+      }
+      deleteRecursively(workDir);
+    } catch (final IOException | InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static class ConfigBuilder {
```

Comment:
Feels like this could be folded into `ArtemisConfiguration`?  You could have a static factory method that could return an `ArtemisConfiguration` instance with the right values, or return a `Builder` with some default values set.  Then have a method on `ArtemisConfiguration` to write the config to a file.

Reply:
It could but I'm keen to keep the acceptance test code separate from the production code.  It does have the downside of adding some extra code, but I've found it extremely helpful at picking up errors where something is named wrong or a stay extra character gets inserted into a string (or even deliberate changes which break backwards compatibility).

I'm envisioning the `ConfigBuilder` growing some fairly high level methods as well so tests are abstracted from the details of what config to write (e.g. a single method to switch between test, normal and simulation deposits that handles the other details, configuring the networking sensibly etc).